### PR TITLE
Emit a clear error instead of crashing on duplicate generated schema names

### PR DESCRIPTION
**Pull request by adityasingh2400**
When the idiomatic naming strategy maps two distinct OpenAPI schema names to the same generated Swift type name, the generator crashes instead of reporting the problem. This is the crash tracked in #854: a document with both `NullTime` and `nullTime` under `#/components/schemas` produces

```
Swift/NativeDictionary.swift: Fatal error: Duplicate values for key: 'NullTime'
```

Root cause is in `boxRecursiveTypes`, which builds a name-to-node lookup for recursion detection with `Dictionary(uniqueKeysWithValues:)`. That initializer traps on duplicate keys, and the idiomatic strategy can produce two declarations sharing a name even though the OpenAPI names were distinct. The defensive strategy keeps them distinct, which is why the crash only shows up with idiomatic naming.

The fix detects the collision before building the lookup and emits an error diagnostic that names the conflicting type and points at the documented workarounds (switch `namingStrategy` to `defensive`, or add a `nameOverrides` entry). The `Configuring-the-generator` article already notes that idiomatic naming "might produce name conflicts (in that case, switch back to defensive)", so this turns an undiagnosed trap into the guidance that already exists. The lookup construction now also tolerates duplicate keys, so the error diagnostic is the failure path rather than a fatal trap, independent of which diagnostic collector is in use.

Verification: running the generator with `namingStrategy: idiomatic` on a spec containing `NullTime` and `nullTime` previously crashed; it now exits non-zero with

```
error: Multiple schemas in '#/components/schemas' map to the same generated Swift type name 'NullTime', which is not supported. ...
```

I added a regression test in `Test_translateSchemas` that translates two schemas colliding under the idiomatic strategy and asserts the error is emitted. With the previous code that test aborts with the `Duplicate values for key` trap (signal 5); with the fix it passes. A second test confirms that distinct idiomatic names emit no error. The full `OpenAPIGeneratorCoreTests` target (115 tests) and the reference tests (114 tests) pass, so generated output for valid documents is unchanged, including recursive-type boxing.

---

Note: this supersedes #907, which I closed by accident during a bulk cleanup of my stale forks and cannot reopen because the original fork was deleted. The branch is identical to the reviewed state of #907 including the uniquingKeysWith rework discussed there.

### CODE REVIEW

File: Sources/_OpenAPIGeneratorCore/Translator/TypesTranslator/translateBoxedTypes.swift [hunk 1]
```diff
@@ -25,7 +25,31 @@ extension TypesFileTranslator {
     func boxRecursiveTypes(_ decls: [Declaration]) throws -> [Declaration] {
 
         let nodes = decls.compactMap(DeclarationRecursionDetector.Node.init)
-        let nodeLookup = Dictionary(uniqueKeysWithValues: nodes.map { ($0.name, $0) })
+        var duplicateNames: Set<String> = []
```
Comment (simonjbeaumont) [inline Sources/_OpenAPIGeneratorCore/Translator/TypesTranslator/translateBoxedTypes.swift]: This PR as-is is a worthwhile, incremental improvement, so we can move forward with this approach. However, if you're motivated to improve it further in a follow up, we can enhance this to keep track of the schema location and even be able to emit diagnostics that show up in the editor, pointing to the duplicate entries in the OpenAPI document.

File: Sources/_OpenAPIGeneratorCore/Translator/TypesTranslator/translateBoxedTypes.swift [hunk 2]
```diff
@@ -25,7 +25,31 @@ extension TypesFileTranslator {
     func boxRecursiveTypes(_ decls: [Declaration]) throws -> [Declaration] {
```
Comment (simonjbeaumont) [inline Sources/_OpenAPIGeneratorCore/Translator/TypesTranslator/translateBoxedTypes.swift:25]: Regardless of whether we enhance to provide a richer diagnostic including schema location, this is not the right place for this check. This function should do only what it says (boxing recursive types).

Further down in this file you'll find `translateSchemas(_:multipartSchemasNames:)`. This logic should move there either as a pass over the computed `[Declaration]` list, or as part of the `flatMap`.

We should also check to see if `TypeAssigner` is a logical place to put this. I can't remember how stateful that is, but if it is stateful, it seems like the best place to keep track of which types have been assigned and emit the diagnostics.

**Comment by adityasingh2400** [comment]
Thanks for the review. I moved the duplicate-name check out of translateBoxedTypes and into translateSchemas, where it now runs as a pass over the computed declarations before boxing. translateBoxedTypes is back to only boxing recursive types, and it keeps the first node on a name collision so it cannot trap if a non-throwing diagnostic collector lets execution continue past the error. The clear error message is unchanged and the existing tests still pass. I left the richer schema-location diagnostic and the TypeAssigner idea as possible follow ups.

File: Sources/_OpenAPIGeneratorCore/Translator/TypesTranslator/translateBoxedTypes.swift [hunk 3]
```diff
@@ -25,7 +25,11 @@ extension TypesFileTranslator {
...
 
         let nodes = decls.compactMap(DeclarationRecursionDetector.Node.init)
-        let nodeLookup = Dictionary(uniqueKeysWithValues: nodes.map { ($0.name, $0) })
+        // Duplicate generated names are detected and reported earlier, in
+        // `translateSchemas`. Keep the first node on a collision here so this
+        // step never traps even if a non-throwing diagnostic collector lets
+        // execution continue past that error.
```
Comment (simonjbeaumont) [inline Sources/_OpenAPIGeneratorCore/Translator/TypesTranslator/translateBoxedTypes.swift:31]:
```suggestion
```

File: Sources/_OpenAPIGeneratorCore/Translator/TypesTranslator/translateSchemas.swift [hunk 4]
```diff
@@ -75,4 +76,33 @@ extension TypesFileTranslator {
...
+            let duplicateName = sortedNames[0]
+            message =
+                "Multiple schemas in '#/components/schemas' map to the same generated Swift type name '\(duplicateName)', which is not supported. This usually happens when the naming strategy collapses two distinct OpenAPI names into one. Switch the namingStrategy to 'defensive', or add a 'nameOverrides' entry to give one of the schemas a different name."
+            context = ["name": duplicateName]
+        } else {
+            let nameList = sortedNames.map { "'\($0)'" }.joined(separator: ", ")
+            message =
+                "Multiple schemas in '#/components/schemas' map to the same generated Swift type names \(nameList), which is not supported. This usually happens when the naming strategy collapses distinct OpenAPI names into one. Switch the namingStrategy to 'defensive', or add 'nameOverrides' entries to give the schemas different names."
+            context = ["names": nameList]
+        }
```
Comment (simonjbeaumont) [inline Sources/_OpenAPIGeneratorCore/Translator/TypesTranslator/translateSchemas.swift:105]: Seems over the top to special case the single entry list. Let's just use one path.
```suggestion
        let nameList = sortedNames.map { "'\($0)'" }.joined(separator: ", ")
        message =
            "Multiple schemas in '#/components/schemas' map to the same generated Swift type names \(nameList), which is not supported. This usually happens when the naming strategy collapses distinct OpenAPI names into one. Switch the namingStrategy to 'defensive', or add 'nameOverrides' entries to give the schemas different names."
        context = ["names": nameList]
```

File: Tests/OpenAPIGeneratorCoreTests/Translator/TypesTranslator/Test_translateSchemas.swift
Comment (simonjbeaumont) [inline Tests/OpenAPIGeneratorCoreTests/Translator/TypesTranslator/Test_translateSchemas.swift:1]: Please add a test that shows that duplicate basenames in different namespaces do not produce an error. E.g. `A.T` and `B.T` are fine. The tests you added show only top-level types in the Schemas dictionary.

**Review by simonjbeaumont — COMMENTED** [review]
Thanks. This is nearly ready to land. Just needs a couple more tests.
